docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+for a given question, that question's contents and index will be copied to a 
+new ArrayList of questions, and subsequently print them.
+
+### Edit Command - Edit Question / Answer
+#### Brief Description of Edit Command
+The CommandEdit class in the quizhub application is responsible for handling user commands to edit the description or 
+answer of a question. The CommandEdit class supports two edit commands: 
+
+- `edit [question number] /description [description]` - edits the description of the question with the specified number
+
+- `edit [question number] /answer [answer]` - edits the answer to the question with the specified number
+
+#### Class Structure of Edit Command
+The CommandEdit class includes the following key components:
+
+- `qnIndex`: An integer representing the question number to be edited.
+- `newDescription`: A string representing the new description for the question (or null if not edited).
+- `newAnswer`: A string representing the new answer for the question (or null if not edited).
+
+#### Implementation of Edit Command
+
+![commandEditSequence.png](UML%2FCommands%2FcommandEditSequence.png)
+
+![commandEditStages.png](UML%2FCommands%2FcommandEditStages.png)
+
+Here are the key steps for implementing this class:
+
+- **Parsing User Input**: Parse the user input to extract the question number, edit criteria (/description or /answer), 
+and new values (description or answer).
 
-## Design & implementation
+- **Validation**: Implement validation logic to ensure that the user input is correctly formatted and contains valid 
+information.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+- **Edit Operation**: Implement the logic to execute the edit operation based on the provided criteria. This typically 
+involves invoking methods in the QuestionList and Question classes to update the question's description or answer.
 
+- **Data Persistence**: If necessary, update the data storage to save the changes. In the provided code, the 
+`dataStorage.updateData(questions)` method is used to save changes to the question list.
 
-## Product scope
-### Target user profile
+- **Error Handling**: Handle any exceptions or errors that may occur during the edit operation and provide appropriate 
+feedback to the user.
 
-{Describe the target user profile}
+### Start Command - Start Quiz
 
-### Value proposition
+#### Brief Description of Start Command
 
-{Describe the value proposition: what problem does it solve?}
+The start quiz feature allows users to start quizzing themselves with customizable characters to define which modules
+to quiz themselves on alongside whether to randomize the questions or use their pre-defined question order.
+- `start /[quiz mode] [start details] /[qn mode]`
+
+#### Class Structure of Start Command
+
+![commandStartSequence.png](UML%2FCommands%2FcommandStartSequence.png)
```

Comment:
Could it be possible to omit the class names as the bottom since it is repeated?
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/23655240/e8d01ddb-f938-492f-996c-9a404f7aa967)